modules/hal_nordic/nrf_802154/CMakeLists.txt

```
@@ -120,3 +120,7 @@ if (CONFIG_NRF_802154_SERIALIZATION)
   zephyr_append_cmake_library(nrf-802154-serialization)
   target_link_libraries(zephyr_interface INTERFACE nrf-802154-serialization-interface)
 endif ()
+
+if (CONFIG_NRF_802154_RADIO_DRIVER OR CONFIG_NRF_802154_SERIALIZATION)
+  target_compile_definitions(zephyr-802154-interface INTERFACE NRF_802154_ENERGY_DETECTED_VERSION=1 )
```

```suggestion
  target_compile_definitions(zephyr-802154-interface INTERFACE NRF_802154_ENERGY_DETECTED_VERSION=1)
```
I'd just move this up, together with other `target_compile_definitions`.

---

done